File: Folicon_Native/Classes/ListItem.vb
Imports System.ComponentModel

Public Class ListItem
    Implements INotifyPropertyChanged
    Public Property Title() As String
    Public Property Year() As String
    Public Property Rating() As String
    Public Property Folder() As String
    Public Property Overview() As String
    Public Property Poster() As String
    Public Sub New(ByVal _title As String, ByVal _year As String, ByVal _rating As String, ByVal Optional _overview As String = Nothing, Optional _poster As String = Nothing, Optional ByVal _folder As String = "")
        Title = _title
        Year = _year
        Rating = _rating
        Overview = _overview
        Poster = _poster
        Folder = _folder
    End Sub

    Public Sub New()
    End Sub

    Protected Sub OnPropertyChanged(ByVal e As PropertyChangedEventArgs)
        RaiseEvent PropertyChanged(Me, e)
    End Sub
    Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged
End Class

File: FoliCon/Models/ListItem.cs
namespace FoliCon.Models;

namespace FoliCon.Models
{
    public class ListItem : BindableBase
    {
        private string _title;
        private string _year;
        private string _rating;
        private string _folder;
        private string _overview;
        private string _poster;
        private string _trailerKey;
        private Uri _trailerLink;
        private int _id;
        public string Title { get => _title; set => SetProperty(ref _title, value); }
        public string Year { get => _year; set => SetProperty(ref _year, value); }
        public string Rating { get => _rating; set => SetProperty(ref _rating, value); }
        public string Folder { get => _folder; set => SetProperty(ref _folder, value); }
        public string Overview { get => _overview; set => SetProperty(ref _overview, value); }
        public string Poster { get => _poster; set => SetProperty(ref _poster, value); }
        public Uri Trailer { get => _trailerLink; set => SetProperty(ref _trailerLink, value); }
        public string TrailerKey { get => _trailerKey; set => SetProperty(ref _trailerKey, value); }
        public int Id { get => _id; set => SetProperty(ref _id, value); }
        public ListItem(string title, string year, string rating, string overview = null, string poster = null, string folder = "", int id = 0)
        {
            Title = title;
            Year = year;
            Rating = rating;
            Overview = overview;
            Poster = poster;
            Folder = folder;
            Id = id;
            
        }

    public ListItem()
    {
    }
}